hooks/init/31-upgrade-puppet.rb

```diff
@@ -1,34 +1,24 @@
-def puppet4_available?
-  success = []
-  success << Kafo::Helpers.execute('yum info puppet-agent')
-  success << Kafo::Helpers.execute('yum info puppetserver')
-  !success.include?(false)
-end
+def puppet5_available?
+  return true if `rpm -q puppet-agent --queryformat=%{version}`.start_with?('5.')
 
-def stop_services
-  Kafo::Helpers.execute('katello-service stop')
-end
+  yum_info = `yum info puppet-agent`
 
-def upgrade_puppet_package
-  Kafo::Helpers.execute('yum remove -y puppet-server')
-  Kafo::Helpers.execute('yum install -y puppetserver')
-  Kafo::Helpers.execute('yum install -y puppet-agent')
-end
+  available = yum_info.split('Available Packages')[1]
+  return false if available.nil?
 
-def start_httpd
-  Kafo::Helpers.execute('katello-service start --only httpd')
+  version = available.split(/Version/)[1].split("\n")[0]
+  version = version.delete(':').strip
+  version.start_with?('5.')
 end
 
-def remove_puppet_port_httpd
-  Kafo::Helpers.execute('sed -i "/^Listen 8140$/d" /etc/httpd/conf/ports.conf')
+def stop_services
+  Kafo::Helpers.execute('katello-service stop')
 end
 
-def copy_data
-  success = []
-  success << Kafo::Helpers.execute('cp -rfp /etc/puppet/environments/* /etc/puppetlabs/code/environments') if File.directory?('/etc/puppet/environments') && !Dir['/etc/puppet/environments/*'].empty?
-  success << Kafo::Helpers.execute('cp -rfp /var/lib/puppet/ssl /etc/puppetlabs/puppet') if File.directory?('/var/lib/puppet/ssl')
-  success << Kafo::Helpers.execute('cp -rfp /var/lib/puppet/foreman_cache_data /opt/puppetlabs/puppet/cache/') if File.directory?('/var/lib/puppet/foreman_cache_data')
-  !success.include?(false)
+def upgrade_puppet_package
+  Kafo::Helpers.execute('yum upgrade -y puppetserver')
+  Kafo::Helpers.execute('yum upgrade -y puppet-agent')
+  Kafo::Helpers.execute('yum reinstall puppet-agent-oauth')
```

Comment:
Got it to go through this reinstall needs to be with a -y so its not hung waiting for input.